main/src/library/pkger/SemVer.flix

```diff
@@ -0,0 +1,118 @@
+namespace SemVer {
+    // TODO: Consider adding pre-release identifier and build metadata
+    pub type alias SemVer = (Int, Int, Int)
+
+    pub enum Ord {
+        case Gt
+        case Eq
+        case Lt
+    }
+
+    pub def isWithinBounds(bound: SemVer, specific: SemVer): Bool = {
+        let majorIsZero = match bound {
+            case (major, _, _) => major == 0
+        };
+
+        if (!majorIsZero) {
```

Comment:
This style of programming is fairly Java-esque or imperative. If you pattern match directly on both semVer you can explicitly match the case when e.g. the major number is zero. I think this would be much prettier as a sequence of pattern matching cases with some guards.